client/components/sticky-panel/test/index.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * @format
+ * @jest-environment jsdom
+ */
+
+/**
+ * External dependencies
+ */
+import sinon from 'sinon';
+import ReactDom from 'react-dom';
+
+import { calculateOffset, getBlockStyle, getDimensions, getDimensionUpdates } from '..';
+
+beforeAll( () => {
+	const header = {
+		getBoundingClientRect: () => ( { height: 123 } ),
+	};
+	sinon.stub( document, 'getElementById' ).callsFake( id => ( id === 'header' ? header : null ) );
+	sinon.stub( ReactDom, 'findDOMNode' ).callsFake( node => node );
```

Comment:
Thanks, @jsnajdr! I usually use Sinon when I want to mock functions in the browser environment, or in modules I'm not responsible for, since it makes it easy to mock, run tests, and restore. Does Jest have similar functionality?

Reply:
Yep! Jest has mocks and spies built in. There's a little bit about it in our [testing docs](https://github.com/Automattic/wp-calypso/blob/master/docs/testing/unit-tests.md), but the [Jest docs](https://jestjs.io/docs/en/getting-started) are quite good.

It looks like you might want something like we have in [our docs](https://github.com/Automattic/wp-calypso/blob/master/docs/testing/unit-tests.md#testing-globals):

```js
jest.spyOn( global, 'open' ).mockImplementation( () => true );
```

Reply:
Thanks, @sirreal, that's perfect! :) I'll reimplement the tests, then.

Reply:
And also for teaching me the right terminology for what I was looking for is "spy" 😄

Reply:
Done.